api/pkg/apis/v1alpha1/utils/parser_test.go

```diff
@@ -2665,6 +2665,68 @@ func TestLeadingUnderScore(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, "a__b", val)
 }
+func TestTriggerGetValue(t *testing.T) {
+	ctx := utils.EvaluationContext{
+		Triggers: map[string]interface{}{
+			"foo":  "bar",
+			"bar":  24,
+			"bazz": true,
+		},
+	}
+	parser1 := NewParser("${{$trigger(foo, -1)}}")
+	val, err := parser1.Eval(ctx)
+	assert.Nil(t, err)
+	assert.Equal(t, "bar", val)
+	parser2 := NewParser("${{$trigger(bar, -1)}}")
+	val, err = parser2.Eval(ctx)
+	assert.Nil(t, err)
+	assert.Equal(t, int(24), val)
+	parser3 := NewParser("${{$trigger(bazz, false)}}")
+	val, err = parser3.Eval(ctx)
+	assert.Nil(t, err)
+	assert.Equal(t, true, val)
```

Comment:
Maybe also get a key not in Triggers map and check the output is default value here.

Reply:
It's in another test `TestTriggerGetDefault`